relengapi/blueprints/slaveloan/tasks.py

```diff
@@ -83,6 +83,33 @@ def choose_inhouse_machine(self, loanid, loan_class):
     return chosen['name']
 
 
+@task(bind=True)
+@add_to_history(
+    before="Identifying aws machine name to use",
+    after="Chose aws machine {retval!s}")
+def choose_aws_machine(self, loanid, loan_class):
+    logger.debug("Choosing aws machine name")
+    # We use foo-$user_shortname$N where $N is optional only if
+    # there exists another active loan with the foo-$user prefix
+    l = Loans.query.get(loanid)
+    prefix = slave_mappings.slavetype_to_awsprefix(loan_class)
+    user_shortname = l.human.ldap.split("@")[0]
```

Comment:
this won't reuse loans, since a number is appended based on current-loan-or-not. While shortnames can conflict in this case, it is still not as common.